azplugins/flow.py

```diff
@@ -717,25 +722,41 @@ def __call__(self, timestep):
 
         if hoomd.comm.get_rank() == 0:
             x = snap.particles.position[:,self.bin_axis]
-            v = snap.particles.velocity[:,self.flow_axis]
+            v = snap.particles.velocity
+            m = snap.particles.mass
 
-            _counts,_ = np.histogram(x, bins=self.bins, range=self.range)
-            self._counts += _counts
+            binids = np.digitize(x, self.edges[1:])
 
-            _velocity,_ = np.histogram(x, bins=self.bins, range=self.range, weights=v)
-            self._velocity += _velocity
+            for dim in range(3):
+               num_vel = np.bincount(binids,v[:,dim],minlength=self.bins)
+               counts = np.bincount(binids,minlength=self.bins)
+               self._mass_velocity[dim] += np.bincount(binids,m*v[:,dim],minlength=self.bins)
+               self._number_velocity[dim] += np.divide(num_vel, counts, out=np.zeros(self.bins), where=counts > 0)
```

Comment:
I think you want to do something a little more like this:
```suggestion
            counts = np.bincount(binids,minlength=self.bins)
            self._counts += counts

            mass = np.bincount(binids,weight=m,minlength=self.bins)
            self._bin_mass += mass

            num_vel = np.zeros((self.bins,3))
            mass_vel = np.zeros((self.bins,3))
            for dim in range(3):
               num_vel[:,dim] = np.bincount(binids,v[:,dim],minlength=self.bins)
               mass_vel[:,dim] = np.bincount(binids,m*v[:,dim],minlength=self.bins)
           np.divide(num_vel, counts, out=num_vel, where=counts > 0)
           np.divide(mass_vel, mass, out=mass_vel, where=mass > 0)
           self._number_velocity[dim] += num_vel
           self._mass_velocity[dim] += mass_vel
```